### PR TITLE
Attempt to kill any stray qemu process

### PR DESCRIPTION
**Pull request by davidcassany**
Signed-off-by: David Cassany <dcassany@suse.com>

### CODE REVIEW

**Comment by davidcassany** [comment]
I realized that exceptions thrown on `LimaBackend.stop()` leave qemu process running behind. This is not necessarily wrong, as for debugging purposes this is great. However while trying to debug #771 I realized the qemu process was still running after closing RD application. `process.kill(0)` is pretty aggressive and even if it works on macOS I believe this is only valid as long as RD runs in its own specific session and I have no idea about how reliable this is for macOS. On linux `process.kill(0)` is just not functional for desktop applications, as most of the desktop applications are sharing the same session.

Because of that I came to this implementation that tries to kill qemu process based on  the `qemu.pid` file that lima produces. I tested this logic in the context of #771

**Comment by jandubois** [comment]
Killing by PID should be fine, but I think `limactl stop -f 0` would be the canonical way to "force quite" the instance when it is in a broken state.

**Comment by davidcassany** [comment]
> Killing by PID should be fine, but I think `limactl stop -f 0` would be the canonical way to "force quite" the instance when it is in a broken state.

Definitely this simplifies even more the logic as I do not have to deal with the pid file and let lima do the job. I was just not aware of this force option in lima :+1:

File: src/k8s-engine/lima.ts [hunk 1]
```diff
@@ -125,16 +125,10 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
...
-    // other running applications under the same gnomeshell session.
-    // If we really need an attempt to kill errant qemu processes a different
-    // strategy is required on Linux.
-    if (!(process.env.NODE_ENV ?? '').includes('test') && !os.platform().startsWith('linux')) {
-      process.on('exit', () => {
+    if (!(process.env.NODE_ENV ?? '').includes('test')) {
+      process.on('exit', async() => {
         // Attempt to shut down any stray qemu processes.
-        process.kill(0);
+        await this.lima('stop', '-f', MACHINE_NAME);
```
Comment (mook-as) [inline src/k8s-engine/lima.ts:131]:
```suggestion
        await this.lima('stop', '--force', MACHINE_NAME);
```
We should always prefer long option names in code / scripts.

**Review by mook-as — CHANGES_REQUESTED** [review]
Trivial change — the code is basically fine (and a lot better than just killing the process group).

File: src/k8s-engine/lima.ts (hunk 1 above)
Comment (davidcassany) [reply src/k8s-engine/lima.ts:131]: Good point, fully agree

**Review by mook-as — APPROVED** [review]

**Pull request merged**